Add spec for CodeLanguage model

timcraft opened the pull request:
Adds specs for the CodeLanguage model (increases coverage from 80% to 100%).

The CodeLanguage#linkable? method is always returning true—if that's correct then the method and the attribute are redundant and could be removed; if not then that's a bug. The `linkable: false` lines in the config suggest the latter but I'm not 100% sure what the correct behaviour is meant to be.

mheap commented:
@timcraft Yep, definitely a bug! `linkable?` should resolve to false for some entries in that file. Could you take a look?

timcraft commented:
@mheap Sure! I've added a commit to fix the bug and updated the spec to match.

```
$ bin/rails console
> CodeLanguageResolver.find('cli').linkable?
=> false
> CodeLanguageResolver.find('curl').linkable?
=> true
```
